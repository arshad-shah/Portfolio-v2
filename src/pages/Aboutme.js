import { Typography, Toolbar, Box } from "@material-ui/core";
import React from "react";
import useStyles from "./styles/aboutMe.styles";
import Picture from "../assets/Picture.webp";
import { aboutMeData } from "../data/aboutMeData";

export default function Aboutme() {
  const classes = useStyles();
  return (
    <Box className={classes.root}>
      <Toolbar className={classes.aboutMeLink} id="aboutme" />
      <Typography align="center" gutterBottom variant="h2">
        About Me
      </Typography>
      <Box className={classes.row}>
        <Box className={classes.aboutMePicture}>
          <img
            src={Picture}
            aria-label="Picture of Arshad shah."
            className={classes.picture}
            width={150}
            height={150}
          />
        </Box>
        <Box className={classes.aboutMeDesc}>
          <Typography variant="body1">
            {aboutMeData.aboutMeDescription}
          </Typography>

          <Typography className={classes.subheading} variant="h5">
            Certificates:
          </Typography>

          <Typography variant="body1">{aboutMeData.certificateName}</Typography>
          <Typography variant="subtitle1">{aboutMeData.grade}</Typography>
          <Typography variant="body2">{aboutMeData.modules}</Typography>
          <Typography variant="subtitle2">{aboutMeData.certDate}</Typography>

          <Typography className={classes.subheading} variant="h5">
            Work Experience:
            <Typography variant="body1">
              {aboutMeData.jobTitle}
              <Typography variant="subtitle1">
                {aboutMeData.jobCompany}
              </Typography>
            </Typography>
            <Typography variant="body2">
              {aboutMeData.jobDescription}
            </Typography>
            <Typography variant="subtitle2">{aboutMeData.jobDate}</Typography>
          </Typography>
        </Box>
      </Box>
    </Box>
  );
}
