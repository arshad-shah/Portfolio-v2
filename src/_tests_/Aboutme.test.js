import React from "react";
import { render, screen } from "@testing-library/react";
import Aboutme from "../pages/Aboutme";

describe("Aboutme of the site:", () => {
  it("should render with all the information in it", () => {
    const { container } = render(<Aboutme />);
    expect(container).toMatchSnapshot();
  });

  it("should render the 'About Me' heading in aboutme", () => {
    render(<Aboutme />);
    const heading = screen.getByText("About Me");
    expect(heading).toBeInTheDocument();
  });
});
