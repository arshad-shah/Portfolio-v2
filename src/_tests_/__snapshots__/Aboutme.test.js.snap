// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aboutme of the site: should render Aboutme with all the information in it 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-10 Aboutme-root-1"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular Aboutme-aboutMeLink-3 MuiToolbar-gutters"
      id="aboutme"
    />
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      About Me
    </h2>
    <div
      class="MuiBox-root MuiBox-root-11 Aboutme-row-4"
    >
      <div
        class="MuiBox-root MuiBox-root-12 Aboutme-aboutMePicture-5"
      >
        <img
          alt="Picture of Arshad shah."
          aria-label="Picture of Arshad shah."
          class="Aboutme-picture-7"
          height="150"
          src="Picture.webp"
          width="150"
        />
      </div>
      <div
        class="MuiBox-root MuiBox-root-13 Aboutme-aboutMeDesc-6"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          I am a student 
          <b>
            Software Engineer
          </b>
           with a passion for building
          <b>
             Beautiful
          </b>
           and 
          <b>
            Intuitive
          </b>
           user interfaces and 
          <b>
            Learning
          </b>
           
          new technologies to building products that 
          <b>
            solve real-world problems
          </b>
          .
          <br />
          I am a 
          <b>
            Team Player
          </b>
           and 
          <b>
            Hard Worker
          </b>
           and I am always
          <b>
             Looking for new challenges
          </b>
          .
          <br />
          I have real world experience in 
          <b>
            Frontend technologies
          </b>
           like
          <b>
             React 
          </b>
          . I have also worked with 
          <b>
            Backend technologies
          </b>
           such as Node.js, Express.js, MongoDB, Firebase, and Firestore as well as with different frameworks such as NextJS, React Native, and Flutter.
          <br />
          I like designing and creating stuff for all domains from 
          <b>
            Mobile
          </b>
           
          to 
          <b>
            Desktop
          </b>
          .
          <br />
          My goal is to make 
          <b>
            Cross Platform
          </b>
          , 
          <b>
            Efficient
          </b>
          , and
           
          <b>
            Clean Application
          </b>
           that give an easy and Perfect experience.
        </p>
        <h5
          class="MuiTypography-root Aboutme-subheading-8 MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Technologies I Use
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              React
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Redux
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Node
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Express
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              MongoDB
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Mongoose
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Jest
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Codeceptjs
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Java
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              HTML
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              CSS
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              TailwindCSS
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              JavaScript
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Ionic
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Capacitor
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Electron
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Firebase
            </span>
          </div>
          <div
            class="MuiChip-root Aboutme-TechChip-9 MuiChip-colorSecondary"
            data-testid="languageChip"
          >
            <span
              class="MuiChip-label"
            >
              Firestore
            </span>
          </div>
        </p>
        <h5
          class="MuiTypography-root Aboutme-subheading-8 MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Work Experience
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          <b>
            Software Engineer
          </b>
           - Intern 
          06/2021 - 08/2021
        </p>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
        >
          <b>
            Houghton Mifflin Harcourt
          </b>
           - Dublin, Ireland
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          During this 
          <b>
            Internship
          </b>
           i worked on a 
          <b>
            Frontend Team
          </b>
           team responsible for the HMH flagship product 
          <b>
            HMH ED
          </b>
          .
          <br />
          Here i learned the 
          <b>
            React framework
          </b>
           using the 
          <b>
            SPA pattern
          </b>
          , unit testing, integration testing, and End to End testing.
          <br />
          I learned crucial skills, following an 
          <b>
             Agile development 
          </b>
          methodology participating in 
          <b>
            sprints
          </b>
          , and completing tickets.
          <br />
          The most important aspect that I learned was the CI pipeline for building complex software and triaging the build process across multiple teams.
        </p>
      </div>
    </div>
  </div>
</div>
`;
